src/Config/Method.php

```diff
@@ -0,0 +1,90 @@
+<?php
+namespace GravApi\Config;
+
+/**
+ * Class Method
+ * @package GravApi\Config
+ */
+class Method
+{
+    /**
+     * @var boolean
+     */
+    protected $enabled = false;
+
+    /**
+     * @var boolean
+     */
+    protected $useAuth = false;
+
+    /**
+     * @var array
+     */
+    protected $fields = array();
+
+    /**
+     * @var array
+     */
+    protected $ignore_files = array();
+
+    public function __construct(array $config = array())
+    {
+        if (isset($config['enabled'])) {
+            $this->enabled = $config['enabled'];
+        }
+
+        if (isset($config['auth'])) {
+            $this->auth = $config['auth'];
+        }
+
+        if (isset($config['fields'])) {
+            // TODO: field string array validation
+            $this->fields = $config['fields'];
+        }
+
+        if (isset($config['ignore_files'])) {
+            // TODO: ignore_files string array validation
+            $this->ignore_files = $config['ignore_files'];
+        }
+    }
+
+    /**
+     * Returns whether or not this endpoint/method is enabled
+     *
+     * @return boolean
+     */
+    public function isEnabled()
+    {
+        return $this->enabled;
+    }
+
+    /**
+     * Returns whether or not this endpoint/method is enabled
+     *
+     * @return boolean
+     */
+    public function useAuth()
+    {
+        return $this->useAuth;
+    }
+
+    /**
+     * Returns whether or not this field should be filtered
+     *
+     * @return boolean
+     */
+    public function shouldFilterField(string $field)
```

Comment:
How will this work with the [existing field filtering setup?](https://github.com/Regaez/grav-plugin-api/blob/master/src/Resources/UserResource.php#L96)

Reply:
Maybe the array should just be exposed instead of a boolean check function?

Reply:
Perhaps the `setFilter` function could be moved into the abstract `Resource` class and made to be generic, e.g.

```php
protected function setFilter()
{
    $filter = Config::instance()->{$this->getResourceType()}->get->fields;
    
    if (!empty($filter)) {
        $this->filter = $filter;
    }
}
```
Then would need to adjust mapping so that `getResourceType()` returns the correct value for the config...

Reply:
I will move this to a separate issue.